.stylelintrc

```diff
@@ -18,7 +18,9 @@
       }
     ],
     "custom-property-empty-line-before": "never",
-    "no-eol-whitespace": null
+    "no-eol-whitespace": null,
+    "selector-nested-pattern": "^(&(:|\\.[\\w-]*--state-)(hover|focus|focus-within|disabled),?\\s*)+$",
```

Comment:
Sadly I can't add comments to json files (maybe that's a reason to make this a `stylelint.config.js`), but this basically only allows certain pseudoclasses (we may want more) and certain classes that communicate a simulated activity state to be nested.

Probably worthwhile to expand to all pseudoclasses and keep the `state-` the same.